dotnet-efcore/UniqueConstraintsInEFCore/UniqueConstraintsInEFCore.Tests/SolarSystemApiApplicationFactory.cs

```diff
@@ -0,0 +1,22 @@
+namespace UniqueConstraintsInEFCore.Tests;
+
+public class SolarSystemApiApplicationFactory : WebApplicationFactory<Program>, IAsyncLifetime
+{
+    private readonly MsSqlContainer _msSqlContainer = new MsSqlBuilder().Build();
+
+    protected override void ConfigureWebHost(IWebHostBuilder builder)
+    {
+        builder.ConfigureServices(services =>
+        {
+            services.RemoveAll(typeof(DbContextOptions<SolarSystemDbContext>));
+            services.AddDbContext<SolarSystemDbContext>(options =>
+                options.UseSqlServer(_msSqlContainer.GetConnectionString()));
+        });
+    }
+
+    public Task InitializeAsync()
+         => _msSqlContainer.StartAsync();
+
+    public new Task DisposeAsync()
```

Comment:
The `DisposeAsync()` method is virtual at least from the documentation perspective. And we should also consider calling the `base.DisposeAsync()` then, since we don't know what is happening under the hood.

Reply:
Ok apparently I didn't see the `IAsyncLifetime `usage, but do we need that when we can override the one from `WebApplicationFactory`?

Reply:
```
public override async ValueTask DisposeAsync() {
    await _msSqlContainer.DisposeAsync();

    await base.DisposeAsync();
}
```

Reply:
This wouldn't work, but I implemented the `DisposeAsync()` from `IAsyncLifetime` explicitly.